packages/analytics-js-plugins/src/iubendaConsentManager/index.ts

```diff
@@ -0,0 +1,126 @@
+      if (!(globalThis as any)._iub?.cs?.consent?.purposes) {
+        state.consents.initialized.value = false;
+        return;
+      }
+
+      try {
+        const allowedConsentIds: string[] = [];
+        const deniedConsentIds: string[] = [];
+
+        // eslint-disable-next-line no-underscore-dangle
+        const { purposes = {} } = (globalThis as any)._iub.cs.api.getPreferences();
+
+        Object.entries(purposes).forEach(e => {
+          const purposeCode = e[0];
+          const isConsented = e[1];
+          if (isConsented) {
+            allowedConsentIds.push(purposeCode);
+          } else {
+            deniedConsentIds.push(purposeCode);
+          }
+        })
+
+        state.consents.initialized.value = true;
+        state.consents.data.value = { allowedConsentIds, deniedConsentIds };
+      } catch (err) {
+        logger?.error(IUBENDA_ACCESS_ERROR(IUBENDA_CONSENT_MANAGER_PLUGIN), err);
+      }
+    },
+
+    isDestinationConsented(
+      state: ApplicationState,
+      destConfig: DestinationConfig,
+      errorHandler?: IErrorHandler,
+      logger?: ILogger,
+    ): boolean {
+      if (!state.consents.initialized.value) {
+        return true;
+      }
+      const allowedConsentIds = state.consents.data.value.allowedConsentIds as string[];
+      const matchPredicate = (consent: string) => allowedConsentIds.includes(consent);
+
+      try {
+        const { iubendaConsentPurposes, consentManagement } = destConfig;
+
+        // If the destination does not have consent management config, events should be sent.
+        if (consentManagement) {
+          // Get the corresponding consents for the destination
+          const cmpConfig = consentManagement.find(c => c.provider === state.consents.provider.value);
+
+          // If there are no consents configured for the destination for the current provider, events should be sent.
+          if (!cmpConfig?.consents) {
+            return true;
+          }
+
+          const configuredConsents = cmpConfig.consents.map(c => c.consent.trim()).filter(n => n);
+          const resolutionStrategy =
+            cmpConfig.resolutionStrategy ?? state.consents.resolutionStrategy.value;
+  
+          // match the configured consents with user provided consents as per
+          // the configured resolution strategy
+          switch (resolutionStrategy) {
+            case 'or':
+              return configuredConsents.some(matchPredicate) || configuredConsents.length === 0;
+            case 'and':
+            default:
+              return configuredConsents.every(matchPredicate);
+          }
+        }
+
+        if (iubendaConsentPurposes) {
+          const configuredConsents = iubendaConsentPurposes.map(p => p.purpose.trim()).filter(n => n);
+
+          // Check if any of the destination's mapped ketch purposes are consented by the user in the browser.
+          return configuredConsents.some(matchPredicate) || configuredConsents.length === 0;
+        }
```

Comment:
This is not required for new CMP integrations. It was only added in other plugins for backward compatibility with legacy consent fields.

Reply:
@mohamedkhattab, what action did you take for this? The whole if block needs to be removed.

Reply:
@saikumarrs ,I did remove it here 7af6d9c, but this I reverted it was causing test failures and I think it won't be a problem since it is for backward compatibility only

Reply:
I have removed this block and updated the tests accordingly.